pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -37,12 +35,20 @@ const (
 // Add registers the webhook for K8s built-in object types.
 func Add(mgr manager.Manager, whiteListedUsers []string, isFleetV1Beta1API bool, denyModifyMemberClusterLabels bool) error {
 	hookServer := mgr.GetWebhookServer()
+	adminGroups := []string{validation.MastersGroup, validation.KubeadmClusterAdminsGroup}
```

Comment:
read the current validation wehbook code to see if there are anything which can be replaced by defining +kubebuilder:validation:XValidation:rule on the api itself

Reply:
check the files under pkg/webhook to see if any validations could be replaced by CEL annotation.

Reply:
only keep the CEL validations changes and add integration tests under test/apis folder

Reply:
#fetch https://kubernetes.io/docs/reference/using-api/cel/ to understand the CEL rules